### PR TITLE
Migrate to material-ui v4

### PR DESCRIPTION
**Pull request by helfi92**
Closes https://github.com/helfi92/material-ui-treeview/issues/36.

### CODE REVIEW

**Comment by helfi92** [comment]
Do you know why this style broke in latest material-ui? It’d be good to know why.

**Comment by Rolikasi** [comment]
I added marginRight: 0 to expandIcon that overrides `.MuiIconButton-edgeEnd` css rule with marginRight: -12px. 
My commit made so many changes in yarn.lock. Is it ok?

**Comment by helfi92** [comment]
> I added marginRight: 0 to expandIcon that overrides .MuiIconButton-edgeEnd css rule with marginRight: -12px.

What I'm trying to understand is why it was working in previous material-ui but broke in material v4.

> My commit made so many changes in yarn.lock. Is it ok?

Yeah, that's okay.

**Comment by Rolikasi** [comment]
> 
> 
> Do you know why this style broke in latest material-ui? It’d be good to know why.

I'm trying to find it in material-ui changelog and documentation. Don't understand why expandIcon gets edgeEnd now. It's too late in my timezone, I'll try to find it tomorrow

**Comment by helfi92** [comment]
Sounds good.

**Comment by Rolikasi** [comment]
[This PR](https://github.com/mui-org/material-ui/pull/14758) added edge props to [IconButton](https://material-ui.com/api/icon-button/#main-content) in material-ui v4, default value is `false`. But expandIcon in [ExpansionPanelSummary](https://material-ui.com/api/expansion-panel-summary/) adds `edge: 'end'` to Icon as default. May be it's material-ui bug )
We can add IconButtonProps={{edge:"false"}} prop to ExpandPanelSummary and we can remove extra classes

**Comment by helfi92** [comment]
> We can add IconButtonProps={{edge:"false"}} prop to ExpandPanelSummary and we can remove extra classes

I like this solution. Let's do it :)

File: src/components/MuiTreeView/index.jsx [hunk 1]
```diff
@@ -205,12 +202,12 @@ function MuiTreeView(props) {
         className={classNames(classes.panel, pickClassName(rest))}>
         <ExpansionPanelSummary
           classes={{
-            expandIcon: classes.expandIcon,
             root: classes.panelSummary,
           }}
           {...expansionPanelSummaryProps}
           className={classNames(pickClassName(expansionPanelSummaryProps))}
           expandIcon={<KeyboardArrowDown />}
+          IconButtonProps={{ edge: 'false' }}
```
Comment (helfi92) [inline src/components/MuiTreeView/index.jsx]: We might want to add this line before `{...expansionPanelSummaryProps}` to allow users to override this object.

**Comment by Rolikasi** [comment]
I'm not sure about padding between items, but it must work after overrides in theme.js.
I can't run `yarn start:styleguide`, cause this warning
theme.js not working in http://localhost:5000/ development server(

warning

**Comment by helfi92** [comment]
Another thing you can try is `rm -rf node_modules/ && yarn && yarn start:styleguide`.

**Comment by Rolikasi** [comment]
>  Another thing you can try is rm -rf node_modules/ && yarn && yarn start:styleguide
> Does yarn run styleguidist server work?

I've tried both, not working. Catch the same error

File: src/components/MuiTreeView/index.jsx [hunk 2]
```diff
@@ -72,6 +72,9 @@ const useStyles = makeStyles(theme => ({
     whiteSpace: 'noWrap',
     maxWidth: '75vw',
   },
+  expandIcon: {
+    right: '8px',
+  },
```
Comment (helfi92) [inline src/components/MuiTreeView/index.jsx]: No need for this class. We can use `edge: 'end'` instead of `false`.

File: src/theme.js [hunk 3]
```diff
@@ -45,16 +45,17 @@ const theme = createMuiTheme({
...
-    unit: 8,
-    double: 16,
-    triple: 24,
-    quad: 32,
-  },
   overrides: {
+    MuiListItem: {
+      root: {
+        paddingTop: '12px',
+        paddingBottom: '12px',
```
Comment (helfi92) [inline src/theme.js]:
```
paddingTop: 12,
paddingBottom: 12,
```

**Comment by helfi92** [comment]
>  I've tried both, not working. Catch the same error

Ah, it's possible that you introduced code that doesn't compile. Check `git status` for any code introduced by accident like an extra comma or something.

**Comment by Rolikasi** [comment]
`import StyleGuide from 'react-styleguidist/lib/**client**/rsg-components/StyleGuide/StyleGuideRenderer';` Works for me

There are two more regressions
**Before**

before

**After** 

after

It seems like after migration css rules work different

css rule

In old version `.makeStyles-panelExpanded` override `.MuiExpansionPanel-root.Mui-expanded`
Should we change it?

**Comment by helfi92** [comment]
> In old version .makeStyles-panelExpanded override .MuiExpansionPanel-root.Mui-expanded
Should we change it?

Yes please :)

**Comment by Rolikasi** [comment]
Is it OK now?

**Comment by helfi92** [comment]
Yeah, looks good :) Do you know why this happened by any chance?

**Comment by Rolikasi** [comment]
React styleguide now uses [Prism for codeviewer](https://react-styleguidist.js.org/docs/cookbook.html#how-to-change-syntax-highlighting-colors) and [these](https://github.com/styleguidist/react-styleguidist/blob/master/src/client/styles/theme.js) values are default
In [demo](https://hassanali.me/material-ui-treeview/) styleguide uses CodeMirror for codeviewer

**Comment by helfi92** [comment]
Awesome work @Rolikasi. Thank you for the help!

**Comment by helfi92** [comment]
Released in material-ui-treeview@4.0.0 🎉

**Comment by Rolikasi** [comment]
Woohoo!!!

**Pull request merged**